portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -31,11 +34,21 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String currcomment = getParameter(request, "comment", "");
-    comments.add(currcomment);
-    response.setContentType("application/json;");
-    String json = convertToJson(comments);
-    response.getWriter().println(json);
+    String currcomment = request.getParameter("comment");
+    System.out.println(currcomment);
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("content", currcomment);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    response.sendRedirect("/index.html");
+
+    // String currcomment = getParameter(request, "comment", "");
+    // comments.add(currcomment);
+    // response.setContentType("application/json;");
+    // String json = convertToJson(comments);
+    // response.getWriter().println(json);
```

Comment:
When you ready with reading from datastore, Move the read part also into a separate method.

private List<String> readComments() {
}

Reply:
Will do